src/java.desktop/share/classes/sun/font/FontManagerFactory.java

```diff
@@ -68,29 +54,9 @@
     @SuppressWarnings("removal")
     public static synchronized FontManager getInstance() {
```

Comment:
The method `FontManagerFactory.getInstance()` is updated to use DLC.

I used synchronization on FontManagerFactory.class in the first check to be consistent with the previous behavior where synchronization was on the `FontManagerFactory.getInstance()` method.

What about to use a static nested class singleton? It could look like:
```
public final class FontManagerFactory {

    public static FontManager getInstance() {
        return FontManagerHolder.instance;
    }

    private static class FontManagerHolder {
        private static final FontManager instance = PlatformFontInfo.createFontManager();
    }
}
```

Reply:
The SunFontManager constructor and its subclasses seem too big, and potentially throw some exceptions and this will ruin the holder idiom since all subsequent calls to this method will fail.

Reply:
I wrote a simple example which uses DLC and lazy initialization holder class idioms. The FontManager is only created when it is accessed the first time via the getInstance() method in both cases and they both fail if an exception is thrown during the class initialization. Are there special restrictions on the class loading time in holder idiom?
```
public class FontManagerFactory {

    private static volatile FontManager instance;

    public static FontManager getInstanceDLC() {

        FontManager result = instance;
        if (result == null) {
            synchronized (FontManagerFactory.class) {
                result = instance;
                if (result == null) {
                    instance = result = new FontManager("One", "Two", "Three");
                }
            }
        }
        return result;
    }

    public static FontManager getInstanceHolder() {
        return FontManagerHolder.instance;
    }


    public static void main(String[] args) {
        String lazyInitializationIdiom = args[0];
        System.out.printf("Use lazy initialization idiom: %s%n", lazyInitializationIdiom);
        if ("DLC".equals(args[0])) {
            System.out.printf("DLC FontManager instance: %s%n", getInstanceDLC());
        } else if ("Holder".equals(args[0])) {
            System.out.printf("Lazy Initialization Holder FontManager instance: %s%n", getInstanceHolder());
        }
    }

    private static class FontManagerHolder {
        private static final FontManager instance = new FontManager("One", "Two", "Three");
    }
}

class FontManager {
    public FontManager(String... args) {
        System.out.println(args[5]);
    }
}
```

Reply:
Does that stack trace difference really matter?
if so, you could wrap the `return FontManagerHolder.instance;` in a try/catch that throws that caused by:

```
try {
  return FontManagerHolder.instance;
} catch (java.lang.ExceptionInInitializerError e) {
  final Throwable cause = e.getCause();
  if (cause instanceof RuntimeException re) {
    throw re;
  }
  //could wrap in IllegalStateException or just throw the error
}
```

If this getInstance method is called with any frequency, getting rid of the volatile access each time is a nice bonus.

Reply:
The difference is that Holder will fail that way forever, while the DLC have a chance to resurrect on next call.

Reply:
So I guess the question is whether there is a valid transitory failure use case?

Reply:
> The difference is that Holder will fail that way forever, while the DLC have a chance to resurrect on next call.

Is it really possible that next call will succeed after the previous one failed with an exception?

Reply:
It may fail when one thread got OOM, but later another/this one will work fine. We can use a holder if there will be some real benefits from using it, but DCL is just simpler and safer in this particular case.